Firestore/core/src/firebase/firestore/util/to_string.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_TO_STRING_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_TO_STRING_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++"
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/util/string_apple.h"
+#include "Firestore/core/src/firebase/firestore/util/string_format.h"
+#include "Firestore/core/src/firebase/firestore/util/type_traits.h"
+#include "absl/meta/type_traits.h"
+#include "absl/strings/str_join.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+template <typename T>
+std::string ToString(const T& value);
```

Comment:
This definition conflicts with the one in [string_util.h](https://github.com/firebase/firebase-ios-sdk/blob/342886739f927a6ddafe60af8d54b69f3d22ca20/Firestore/core/src/firebase/firestore/util/string_util.h#L72). We should reconcile these.

Reply:
Good catch, I completely forgot about this one. I think the new definition has exactly the same behavior, but is more powerful, so the previous one can be deleted. What do you think?